src/rulesets/properties.ts

```diff
@@ -23,9 +24,35 @@ function withinAttributes(context) {
   return false;
 }
 
+function bodyPropertyName(context) {
+  const prefix =
+    "inRequest" in context
+      ? "request"
+      : "inResponse" in context
+      ? "response"
+      : "";
+  return "jsonSchemaTrail" in context
+    ? `${prefix} property ${context.jsonSchemaTrail.join(".")}`
+    : `${prefix} property`;
+}
+
+const preventChange = (property: string) => {
+  return (parameterBefore, parameterAfter, context) => {
+    let beforeSchema = (parameterBefore.flatSchema ||
+      {}) as OpenAPIV3.SchemaObject;
+    let afterSchema = (parameterAfter.flatSchema ||
+      {}) as OpenAPIV3.SchemaObject;
+    if (!beforeSchema[property] && !afterSchema[property]) return;
+    expect(
+      beforeSchema[property],
+      `expected ${bodyPropertyName(context)} ${property} to not change`,
+    ).to.equal(afterSchema[property]);
```

Comment:
Dropping a constraint, going from more restrictive, to less restrictive, should be OK. So in the case of the `minimum` and `maximum` attributes, if you removed them, that would be non-breaking, because existing clients will still satisfy the looser constraints.

I don't think we need to do exhaustive type analysis on changes in `type`, `format` or `pattern` to detect a compatible looser constraint and allow it; it's fine to keep things simple for these and just not allow them to change within a version.

Reply:
Ok... I think I see what's going on here. `property` passed in here, is the schema property that is not allowed to change. Perhaps `schemaProperty` would be a clearer name for this? Or a jsdoc comment on function?

Reply:
Agreed. I'll change it up.